### PR TITLE
ONE: Added card "Eye of Malcator" and its abilities and effects

### PR DESCRIPTION
**Pull request by Merlingilb**

### CODE REVIEW

File: Mage.Sets/src/mage/cards/e/EyeOfMalcator.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import mage.game.permanent.token.custom.CreatureToken;
+
+import java.util.UUID;
+
+public class EyeOfMalcator extends CardImpl {
+    public EyeOfMalcator(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{2}{U}");
+
+        //When Eye of Malcator enters the battlefield, scry 2.
+        this.addAbility(new EntersBattlefieldAbility(new ScryEffect(2)));
```
Comment (PurpleCrowbar) [inline Mage.Sets/src/mage/cards/e/EyeOfMalcator.java:24]: This needs to use an instance of `EntersBattlefieldTriggeredAbility`, not `EntersBattlefieldAbility`

File: Mage.Sets/src/mage/cards/e/EyeOfMalcator.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+        //Whenever another artifact enters the battlefield under your control, Eye of Malcator becomes a 4/4 Phyrexian
+        //Eye artifact creature until end of turn.
+        this.addAbility(new EntersBattlefieldControlledTriggeredAbility(new BecomesCreatureSourceEffect(
```
Comment (PurpleCrowbar) [inline Mage.Sets/src/mage/cards/e/EyeOfMalcator.java:28]: Please apply `.setText("{this} becomes a 4/4 Phyrexian Eye artifact creature until end of turn")` to `BecomesCreatureSourceEffect` for desired rule text

File: Mage/src/main/java/mage/filter/StaticFilters.java [hunk 3]
```diff
@@ -407,6 +407,14 @@ public final class StaticFilters {
         FILTER_CONTROLLED_ARTIFACT_OR_OTHER_CREATURE.setLockedFilter(true);
     }
 
+    public static final FilterControlledPermanent FILTER_CONTROLLED_ANOTHER_ARTIFACT = new FilterControlledPermanent("another artifact");
```
Comment (PurpleCrowbar) [inline Mage/src/main/java/mage/filter/StaticFilters.java:410]: The name text should be "another artifact you control"

**Review by PurpleCrowbar — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/e/EyeOfMalcator.java (hunk 1 above)
Comment (Merlingilb) [reply Mage.Sets/src/mage/cards/e/EyeOfMalcator.java:24]: done

File: Mage.Sets/src/mage/cards/e/EyeOfMalcator.java (hunk 2 above)
Comment (Merlingilb) [reply Mage.Sets/src/mage/cards/e/EyeOfMalcator.java:28]: done

File: Mage/src/main/java/mage/filter/StaticFilters.java (hunk 3 above)
Comment (Merlingilb) [reply Mage/src/main/java/mage/filter/StaticFilters.java]: done

**Pull request merged**